Algorithms/GEMETD.py

```diff
@@ -0,0 +1,56 @@
+from Algorithms.ETDLB import ETDLB
+from Algorithms.BaseTD import BaseTD
+import numpy as np
+
+
+class GEMETD(ETDLB):
+    """
+    An ETD(0) implementation that uses GEM (aka GTD2(0) with x and x_p switched) to estimate emphasis.
+    """
+    def __init__(self, task, **kwargs):
+        super().__init__(task, **kwargs)
+        self.beta = self.task.GAMMA
+        self.gem_alpha = self.get('gem_alpha')  # Step size for GEM weights.
+        self.gem_eta = self.get('gem_eta')  # Regularization parameter for GEM; not needed for a fixed target policy.
+        self.k = np.zeros(self.task.num_features)  # Auxiliary weights for GTD2.
+        self.u = np.zeros(self.task.num_features)  # Main weights for GTD2.
+        if self.task.num_policies > 1:
+            self.k = np.zeros((self.task.num_policies, self.task.num_features))
+            self.u = np.zeros((self.task.num_policies, self.task.num_features))
+
+    @staticmethod
+    def related_parameters():
+        return ['alpha', 'gem_alpha', 'gem_eta']
+
+    def learn_single_policy(self, s, s_p, r, is_terminal):
+        x, x_p = self.get_features(s, s_p, is_terminal)
+        rho = self.get_isr(s)
+        # GEM update (regularized GTD2(0) with x and x_p switched):
+        delta_bar = 1 + rho * self.gamma * np.dot(self.u, x) - np.dot(self.u, x_p)
+        self.k += self.gem_alpha * (delta_bar - np.dot(self.k, x_p)) * x_p
```

Comment:
This probably should use the second step size parameter? Like I used ``alpha_v=eta*self.alpha`` in Gradient-TD algorithms?

Reply:
For some reason the GEM paper uses the same step size for both emphasis weight vectors `self.k` and `self.u`. This seems questionable for the same reasons it's questionable in Gradient-TD algorithms, but it's what they went with.